UndertaleModLib/Decompiler/Disassembler.cs

```diff
@@ -9,30 +9,36 @@ namespace UndertaleModLib.Decompiler
 {
     public static class Disassembler
     {
-        public static string GenerateLocalVarDefinitions(this UndertaleCode code, IList<UndertaleVariable> vars, UndertaleCodeLocals locals)
+        private static void AppendLocalVarDefinitionsToStringBuilder(StringBuilder sb, UndertaleCode code, IList<UndertaleVariable> vars, UndertaleCodeLocals locals)
         {
             if (code.WeirdLocalFlag)
-                return "";
-            if (locals == null)
-                return "; Missing code locals- possibly due to unsupported bytecode version or brand new code entry.\n";
-
-            StringBuilder sb = new StringBuilder();
+            {
+                return;
+            }
 
+            if (locals is null)
+            {
+                sb.AppendLine("; Missing code locals, possibly due to unsupported bytecode version or brand new code entry.");
+                return;
+            }
+            
             var referenced = code.FindReferencedLocalVars();
             if (locals.Name != code.Name)
-                throw new Exception("Name of the locals block does not match name of the code block");
+                throw new Exception("Name of the locals block does not match name of the code block!");
             foreach (var arg in locals.Locals)
             {
-                sb.Append(".localvar " + arg.Index + " " + arg.Name.Content);
-                var refvar = referenced.Where((x) => x.Name == arg.Name && x.VarID == arg.Index).FirstOrDefault();
-                if (refvar != null)
+                sb.Append(".localvar ");
+                sb.Append(arg.Index);
+                sb.Append(' ');
+                sb.Append(arg.Name.Content);
```

Comment:
I take it that interpolated strings are parsed the same as `+` concatenation, meaning this separation is the only way that improves performance?

Reply:
Actually... not 100% sure. I know interpolation is faster than concatenation. But I haven't actually explicitly tested whether interpolation would be much slower here than seperate calls.
I assumed that seperate calls were faster than interpolation, because 
a) interpolation does some formatting still. While here, it can choose the most optimal path for the constant/int/char/string.
b) interpolation more or less still string-ifies all the individual elements.